### PR TITLE
Polishing

### PR DESCRIPTION
**Pull request by raphael-theriault-swi**
Bunch of small things that I was planning to push with the mysql update but at this point they're more than worth their own PR. The biggest change is the upgrade from Yarn 3 to 4, everything else is detailed in comments.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,3 +1 @@
 yarn.lock linguist-generated
-.yarn/releases/** filter=lfs diff=lfs merge=lfs -text linguist-generated
-.yarn/plugins/** filter=lfs diff=lfs merge=lfs -text linguist-generated
```
Comment (raphael-theriault-swi) [inline .gitattributes:3]: Yarn 4 comes with a lot more plugins bundled and uses Node's corepack for version/release management.

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # Ignore yarn cache except yarn itself and its plugins
 .yarn/*
-!.yarn/plugins/
-!.yarn/releases/
```
Comment (raphael-theriault-swi) [inline .gitignore:4]: See comment in `.gitattributes`

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,6 @@
...
+!.yarn/versions/
```
Comment (raphael-theriault-swi) [inline .gitignore:3]: Used to be ignored which breaks the version bump stuff.

File: .yarnrc.yml [hunk 4]
```diff
@@ -1,16 +1,18 @@
-plugins:
-  - path: .yarn/plugins/@yarnpkg/plugin-version.cjs
-    spec: "@yarnpkg/plugin-version"
-  - path: .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs
-    spec: "@yarnpkg/plugin-interactive-tools"
-
-yarnPath: .yarn/releases/yarn-3.6.0.cjs
```
Comment (raphael-theriault-swi) [inline .yarnrc.yml:7]: No longer necessary with Yarn 4

File: examples/package.json [hunk 5]
```diff
@@ -2,12 +2,12 @@
   "name": "@solarwinds-apm/examples",
   "private": true,
   "scripts": {
-    "lint": "prettier --check . && eslint --max-warnings=0 *.js '*/**/*.{js,ts}'",
-    "lint:fix": "eslint --fix *.js '*/**/*.{js,ts}' && prettier --write ."
```
Comment (raphael-theriault-swi) [inline examples/package.json:6]: Realised eslint got smarter and I don't have to specify the files to lint manually anymore :D

File: package.json [hunk 6]
```diff
@@ -24,9 +24,14 @@
...
     "prettier": "^3.0.0",
-    "turbo": "^1.10.7",
+    "turbo": "^1.10.10",
     "typescript": "^5.1.6"
+  },
+  "dependenciesMeta": {
+    "prettier": {
+      "unplugged": true
+    }
   }
```
Comment (raphael-theriault-swi) [inline package.json:36]: Required for Prettier 3.0 to work properly with VSCode

File: packages/eslint-config/index.js
Comment (raphael-theriault-swi) [inline packages/eslint-config/index.js:1]: Upgraded to `typescript-eslint` 6 which slightly changes their bundled configs and available options.

File: packages/sdk/src/patches/fs.ts
Comment (raphael-theriault-swi) [inline packages/sdk/src/patches/fs.ts:1]: Fixes the `fs` root span spam seen in many web frameworks. This has been a long standing issue with the current library and dependents can still manually turn on the option if they want to.

File: packages/sdk/src/sampler.ts [hunk 7]
```diff
@@ -232,11 +231,6 @@ export class SwoSampler implements Sampler {
     parentSpanContext: SpanContext | undefined,
     traceOptions: TraceOptions | undefined,
   ): TraceState {
-    traceState = traceState.set(
-      TRACESTATE_SW_KEY,
-      swValue(parentSpanContext, decisions),
-    )
```
Comment (raphael-theriault-swi) [inline packages/sdk/src/sampler.ts:238]: Unnecessary as illustrated by @tammy-baylis-swi

File: scripts/version.js [hunk 8]
```diff
@@ -80,6 +80,9 @@ exec("git add yarn.lock 'packages/**/package.json'")
 const bumped = gitStatus()
   .filter(({ file }) => /packages\/[^/]+\/package.json/.test(file))
   .map(({ file }) => readJson(file))
+
+exec("git commit -m 'versions'")
```
Comment (raphael-theriault-swi) [inline scripts/version.js:84]: Required for the created tags not to point to the previous commit.

File: scripts/vscode.js [hunk 9]
```diff
@@ -36,6 +36,13 @@ for (const recommendation of recommendations) {
 if (!settingsJson["eslint.experimental.useFlatConfig"]) {
   settingsJson["eslint.experimental.useFlatConfig"] = true
 }
+const eslintWorkingDirectories = ["packages/*", "examples", "scripts"]
+settingsJson["eslint.workingDirectories"] ??= eslintWorkingDirectories
+for (const ewd of eslintWorkingDirectories) {
+  if (!settingsJson["eslint.workingDirectories"].includes(ewd)) {
+    settingsJson["eslint.workingDirectories"].push(ewd)
+  }
+}
```
Comment (raphael-theriault-swi) [inline scripts/vscode.js:45]: Required for the ESLint integration to work properly, forgot everyone wouldn't have it set up.

**Review by tammy-baylis-swi — APPROVED** [review]
Yay sampler cleanup 😺

**Comment by raphael-theriault-swi** [comment]
Wish there was a way to not have all of the packages have to be fully specified but it's not possible with any package package manager as far as I'm aware.

**Pull request merged**